verilog/tools/ls/symbol-table-handler.cc

```diff
@@ -0,0 +1,267 @@
+  return uri.substr(kFileSchemePrefix.size());
+}
+
+std::string PathToLSPUri(absl::string_view path) {
+  std::filesystem::path p(path.begin(), path.end());
+  return absl::StrCat(kFileSchemePrefix, std::filesystem::absolute(p).string());
+}
+
+void SymbolTableHandler::SetProject(
+    const std::shared_ptr<VerilogProject> &project) {
+  curr_project_ = project;
+  ResetSymbolTable();
+  if (curr_project_) LoadProjectFileList(curr_project_->TranslationUnitRoot());
+}
+
+void SymbolTableHandler::ResetSymbolTable() {
+  symbol_table_ = std::make_unique<SymbolTable>(curr_project_.get());
+}
+
+std::vector<absl::Status> SymbolTableHandler::BuildSymbolTableFor(
+    const VerilogSourceFile &file) {
+  return BuildSymbolTable(file, symbol_table_.get(), curr_project_.get());
+}
+
+std::vector<absl::Status> SymbolTableHandler::BuildProjectSymbolTable() {
+  ResetSymbolTable();
+  if (!curr_project_) {
+    return {absl::UnavailableError("VerilogProject is not set")};
+  }
+  LOG(INFO) << "Parsing project files...";
+  std::vector<absl::Status> buildstatus;
+  symbol_table_->Build(&buildstatus);
+  for (const auto &diagnostic : buildstatus) {
+    LOG(WARNING) << diagnostic.message();
+  }
+  std::vector<absl::Status> resolvestatus;
+  symbol_table_->Resolve(&resolvestatus);
+  for (const auto &diagnostic : resolvestatus) {
+    LOG(WARNING) << diagnostic.message();
+  }
+  buildstatus.insert(buildstatus.end(), resolvestatus.begin(),
+                     resolvestatus.end());
+  files_dirty_ = false;
+  return buildstatus;
+}
+
+void SymbolTableHandler::LoadProjectFileList(absl::string_view current_dir) {
+  LOG(INFO) << __FUNCTION__;
+  if (!curr_project_) return;
+  // search for FileList file up the directory hierarchy
+  FileList filelist;
+  std::string projectpath;
+  if (auto status = verible::file::UpwardFileSearch(
+          current_dir, absl::GetFlag(FLAGS_file_list_path), &projectpath);
+      !status.ok()) {
+    LOG(WARNING) << "Could not find " << absl::GetFlag(FLAGS_file_list_path)
+                 << " file in the project:  " << status;
+    return;
+  }
+  LOG(INFO) << "Found file list under " << projectpath;
+  // fill the FileList object
+  if (absl::Status status = AppendFileListFromFile(projectpath, &filelist);
+      !status.ok()) {
+    // if failed to parse
+    LOG(WARNING) << "Failed to parse file list in " << projectpath << ":  "
+                 << status;
+    return;
+  }
+  // add directory containing filelist to includes
+  // TODO (glatosinski): should we do this?
+  curr_project_->AddIncludePath(verible::file::Dirname(projectpath));
+  LOG(INFO) << "Adding \"" << projectpath << "\" to include directories";
+  // update include directories in project
+  for (const auto &incdir : filelist.preprocessing.include_dirs) {
+    LOG(INFO) << "Adding include path:  " << incdir;
+    curr_project_->AddIncludePath(incdir);
+  }
+  // add files from file list to the project
+  for (auto &incfile : filelist.file_paths) {
```

Comment:
these are not included files, just files that are part of the project. So `incfile` is not necessarily a good name. How about `file_in_project` ?

Also `OpenIncludedFile()` is of course not necessarily the right name for the function, but maybe add that as TODO to provide a better name. It also goes through the include paths and attempts to find them relative to that. which is not neccesarily what we need. We do want to find them relative to project root.
Maybe add a TODO what needs to be improved later.